python/ray/ml/checkpoint.py

```diff
@@ -0,0 +1,112 @@
+import abc
+
+import ray
+
+
+class Checkpoint(abc.ABC):
+    """Checkpoint interface"""
+
+    @classmethod
+    def from_bytes(cls, data: bytes) -> "Checkpoint":
+        """Create checkpoint object from bytes stream.
+
+        Args:
+            data (bytes): Data stream containing pickled checkpoint data.
+
+        Returns:
+            Checkpoint: checkpoint object.
+        """
+        raise NotImplementedError
+
+    def to_bytes(self) -> bytes:
```

Comment:
We'll probably want to support streams (e.g., to_bytes(file_like), from_bytes(file_like)), maybe add a TODO?